roles/satellite-clone/tasks/main.yml

```diff
@@ -190,17 +169,33 @@
   command: restorecon -R /
   when: restorecon
 
-# Run Satellite installer
-- name: run Satellite 6.1 installer
+# This file tells the candlepin puppet module in the Satellite installer that
+# candlepin is already set up. We remove it so candlepin is set up correctly.
+- name: Remove cpdb_done file
+  file:
+    path: /var/lib/candlepin/cpdb_done
+    state: absent
+
+- name: Run Satellite 6.1 installer
   command: katello-installer --capsule-dns false --capsule-dhcp false --capsule-tftp false
-  when: satellite_version == 6.1
-- name: run Satellite 6.2 installer
+  when: satellite_version == "6.1"
+
+- name: "Run Satellite 6.2 installer"
   command: satellite-installer --scenario satellite --foreman-proxy-dns false --foreman-proxy-dhcp false --foreman-proxy-tftp false --foreman-ipa-authentication false
-  when: satellite_version == 6.2
+  environment:
+    HOSTNAME: "{{ hostname }}"
```

Comment:
what is the need for this?

Reply:
So this is something that only occurs in 6.3, but I added in 6.2 as well to be safe. 

The $HOSTNAME bash variable doesn't update when running the playbook after we change the hostname. Some of the puppet modules that the installer runs rely on this bash variable. In this case it was making calls to the foreman-proxy using $HOSTNAME, but using the old $HOSTNAME bash variable, before it was changed to match the backup's hostname. This ensures that the bash variable is updated.

Reply:
I wonder if [`meta: reset_connection`](https://docs.ansible.com/ansible/latest/meta_module.html) would help here and be less of a hack